network-api/networkapi/homepage/tests/tests_views.py

```diff
@@ -0,0 +1,67 @@
+from datetime import datetime, timedelta, timezone
+
+from django.test import TestCase
+from unittest import expectedFailure
+from rest_framework.test import APIRequestFactory
+from networkapi.homepage.views import HomepageView
+from networkapi.homepage.factory import (
+    HomepageFactory,
+    HomepageLeadersFactory,
+    HomepageNewsFactory,
+    HomepageHighlightsFactory,
+)
+
+
+class TestHomepageView(TestCase):
+    """
+    Test HomepageView class
+    """
+
+    def setUp(self):
+        self.factory = APIRequestFactory()
+
+    def test_homepage_view_response_code(self):
+        """
+        Make sure the homepage returns a 200 status code
+        """
+
+        homepage = HomepageFactory()
+        HomepageLeadersFactory(homepage=homepage)
+        HomepageNewsFactory(homepage=homepage)
+        HomepageHighlightsFactory(homepage=homepage)
+
+        request = self.factory.get('/api/homepage/')
+        response = HomepageView.as_view()(request)
+
+        self.assertEqual(response.status_code, 200)
+
+    def test_homepage_view_404(self):
+        """
+        Make sure HomepageView returns a 404 if there's no Homepage
+        """
+
+        request = self.factory.get('/api/homepage/')
+        response = HomepageView.as_view()(request)
+
+        self.assertEqual(response.status_code, 404)
+
+    @expectedFailure
```

Comment:
I'm not sure how the expired stuff is supposed to work for the homepage but let's at least file an issue (and link it here) to make sure that someone looks at it

Reply:
I think the issue that @patjouk filed covers this? #958